packages/next/build/index.ts

```diff
@@ -2037,6 +2037,7 @@ export default async function build(
       for (const file of [
         ...requiredServerFiles.files,
         path.join(config.distDir, SERVER_FILES_MANIFEST),
+        '.env',
```

Comment:
We can use the [loadedEnvFiles variable](https://github.com/vercel/next.js/blob/28f65ff4116824d888f0b29c17b08173a27d751f/packages/next/build/index.ts#L142) here although we need to filter any that end in `.local`, this ensures `.env.production` is copied as well

Reply:
I also added a generic to `traceSync` so it passes through the return type of the callback. Was easier to see what `loadedEnvFiles` looks like.